tests/pipeline_operations_test.py

```diff
@@ -72,14 +72,107 @@ def test_reduce_accumulators_per_key(self):
                                                            (8, 1)])
             col = self.ops.map_values(col, SumAccumulator,
                                       "Wrap into accumulators")
-            col = self.ops.reduce_accumulators_per_key(col)
+            col = self.ops.reduce_accumulators_per_key(
+                col, "Reduce accumulators per key")
             result = col | "Get accumulated values" >> beam.Map(
                 lambda row: (row[0], row[1].get_metrics()))
 
             beam_util.assert_that(result,
                                   beam_util.equal_to([(6, 2), (7, 2), (8, 1)]))
 
 
+class BeamOperationsStageNameTest(unittest.TestCase):
+
+    def test_beam_create_stage_name_must_be_unique(self):
+        with test_pipeline.TestPipeline() as p:
+            p | f"SAME_BEAM_CREATE_NAME" >> beam.Create([(6, 1), (6, 2)])
+            with self.assertRaisesRegex(
+                    RuntimeError,
+                    expected_regex="A transform with label "
+                    "\"SAME_BEAM_CREATE_NAME\" already exists "
+                    "in the pipeline."):
+                p | f"SAME_BEAM_CREATE_NAME" >> beam.Create([(6, 1), (6, 2)])
+
+    def test_ops_stage_name_must_be_unique(self):
```

Comment:
I have a suggestion to restructure tests in the following way.

1.To have unit tests for `UniqueLabelGenerators`
2. To test all Beam function by mocking everything what's needed, i.e. w/o setup pipelines just to test that  `UniqueLabelGenerators` is called in all places like  `col | ulg.unique_label >> transform`

Tests for `UniqueLabelGenerators ` are just standard unittests. 
I'll explain how to do mocking for `map`:
1. Introduce `MockUniqueLabelGenerators:

```
 class MockUniqueLabelGenerators:
         def unique():
              return "unique_label"
```
2. Introduce MockPcollection
```
from unittest.mock import Mock, MagicMock, patch
def create_mock_pcollection():
    mock = Mock()
    mock.__or__ = MagicMock(return_value = mock)
    return mock
```
3.Testing of Map
 ```
 @patch("apache_beam.transforms.ptransform.PTransform.__rrshift__")
  def test_map(self, mock_rrshift):
        mock_pcollection = create_mock_pcollection()
        ops = BeamOperations(MockUniqueLabelGenerators())
        ops.map(mock_pcollection, lambda x:x, "stage_name")
        mock_rrshift.assert_called_once_with("unique_value")
```

Explanation for patch:
\@patch  creates mock for `>>` in expression   col | "stage_name" >> beam.Map(fn), and in line
` mock_rrshift.assert_called_once_with("unique_value")` we check that the argument of `>>` is acctually value returned from MockUniqueLabelGenerator

WDYT?

Please do it for now only for `map` and then in next step we will generalize it for all methods

Reply:
Awesome! I added tests for all the functions in Beam Pipeline operations.
WDYT about adding 'MockUniqueLabelGenerators' and `create_mock_pcollection` to the `BeamOperationsStageNameTest` class?

Reply:
@dvadym Not sure why, as I recently merged main into this branch, but one of the test fails starting this morning: https://github.com/OpenMined/PipelineDP/runs/4777317330
Is it OK?

Reply:
Thanks a lot! 

It looks like this test fails because of some Spark internal stuff. I'll investigate it later, it's independent of the current PR.

Reply:
Thanks for looking into this! 🚀